shared-module/displayio/Group.c

```diff
@@ -358,33 +374,35 @@ void displayio_group_construct(displayio_group_t *self, mp_obj_list_t *members,
 bool displayio_group_fill_area(displayio_group_t *self, const _displayio_colorspace_t *colorspace, const displayio_area_t *area, uint32_t *mask, uint32_t *buffer) {
     // Track if any of the layers finishes filling in the given area. We can ignore any remaining
     // layers at that point.
-    for (int32_t i = self->members->len - 1; i >= 0; i--) {
-        mp_obj_t layer;
-        #if CIRCUITPY_VECTORIO
-        const vectorio_draw_protocol_t *draw_protocol = mp_proto_get(MP_QSTR_protocol_draw, self->members->items[i]);
-        if (draw_protocol != NULL) {
-            layer = draw_protocol->draw_get_protocol_self(self->members->items[i]);
-            if (draw_protocol->draw_protocol_impl->draw_fill_area(layer, colorspace, area, mask, buffer)) {
-                return true;
+    if (self->hidden == false) {
```

Comment:
Seems like this will break shapes cleaning up after they’re set hidden, no?

Reply:
I'm not sure. I still don't quite have a full picture in my mind of how everything works together. But I'm picking up new pieces each time I dive in and attempt changes like this one so I'm getting closer.

I haven't noticed any unwanted effects of being done this way, however the only test scripts I've used are the relatively basic one in the issue #6204 and more complex one that is a modified version of the TabLayout example from https://github.com/adafruit/Adafruit_CircuitPython_DisplayIO_Layout/pull/70 In both of those cases the hiding and showing seems to work as intended.

By cleaning up do you mean like garbage collection? or something different? Do you have in mind some "stress test" steps that I could try in order to see if it's problematic?

Reply:
Yeah, if you move the shape/anything then set hidden=true within 1 draw’s interval I think you will regress all displayio objects’ cleanups with this change if I understand and remember correctly (which is not a guarantee)

Reply:
I tested a few variations of changing objects while the group they belong to was hidden. Here is the base test script:
```py
import time
import displayio
import terminalio
import vectorio
from adafruit_display_text.bitmap_label import Label
import board

display = board.DISPLAY

display.auto_refresh = False

main_group = displayio.Group()

palette = displayio.Palette(1)
palette[0] = 0x125690

circle = vectorio.Circle(pixel_shader=palette, radius=25, x=70, y=40)
main_group.append(circle)

rectangle = vectorio.Rectangle(pixel_shader=palette, width=40, height=30, x=155, y=45)
main_group.append(rectangle)

lbl = Label(terminalio.FONT, scale=3, text="Hello Shapes!", anchor_point=(0,0), anchored_position=(100, 100))

main_group.append(lbl)

display.show(main_group)
display.refresh()


main_group.hidden = True
time.sleep(2)
display.refresh()
# lbl.y += 100
circle.radius = 45
lbl.background_color = 0xdd00dd
lbl.x += 55
main_group.y += 100
main_group.pop()

time.sleep(1)
main_group.hidden = False
display.refresh()

time.sleep(1)
main_group.append(lbl)
display.refresh()

while True:
    pass
```
I ran it with different variations of order and changes various aspects of the different objects.

In all scenarios that I observed everything seems to react as expected. The changes end up taking effect and being visible once the group is shown again (and the display is refreshed).

I'm a little unsure what you mean by "cleanups" in this context. Do you mean something about being garbage collected by the system? or cleanup in the "dirty" display sense i.e. un-refreshsed? Or something else?

Reply:
The “cleanup” I’m talking about is the “dirty area” left behind when a displayio object has changed location. It’s tracked by the “dirty area” and ends up going through a draw on the next frame. The object is supposed to not draw areas that it is no longer interested in, which leaves the bit flag byte array region covering those areas as `0` which means “this shape didn’t set those pixels’ values.”

I might be misremembering the lifecycle here in group - it has been a while!

The thing I’m concerned about is usually visible with motion. Something like this will tickle a few cleanup cases that have caught me out before:

```
while True:
  circle.x = (circle.x + 1) % 240
  circle.y = (circle.y + 2) % 320
  main_group.hidden = True
  time.sleep(0.004)
  circle.x = (circle.x + 1) % 240
  circle.y = (circle.y + 2) % 320
  main_group.hidden = False
  time.sleep(0.004)
```

Reply:
I tried this out and tweaked it a bit to match the Titano display size. 

https://user-images.githubusercontent.com/2406189/168932620-c4ccb417-963f-4703-90c3-6f5eae2f10ce.mp4

It's behavior seems normal / expected to me. It's quite "flashy" because of the short sleeps and being made hidden / visible but I think that is expected with this code. 

The camera feed / streamed recording it came from has glitch some during the video but the display itself doesn't have the same pause glitch in it.

I don't spot anything problematic with it right away. I could try out any other examples you can think of to expose related issues.

Reply:
Okay, cool! This looks perfect; seems like I’m misremembering. If it was as I thought I remembered then the problem would be plain to see with this tester. Thank you kindly for humoring my uncertainty.

Reply:
@WarriorOfWire For sure, I'm happy to do additional testing for edge cases to make sure I've not broken anything. Thank you for having a look and pointing me toward the right way to get this accomplished! I probably would have gone "off the rails" for a few parts of this change without your guidance and point in the right direction. Much appreciated!